### PR TITLE
feat: WalletAccount non-breaking temp solution

### PR DESCRIPTION
**Pull request by tabaktoni**
## Motivation and Resolution

Based on #1245
This should be a temporary fix that can be merged as it should not contain breaking changes.

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Comment by PhilippeR26** [comment]
What are the differences with PR1245?

**Comment by tabaktoni** [comment]
> What are the differences with PR1245?

Please check now, I started PR but got pulled on HH

**Comment by tabaktoni** [comment]
In general, the idea is to have a non-breaking constructor solution. So for existing users, this should continue to work as it was before, but now they can also provide an address or use static methods to avoid wallet popup.

**Comment by PhilippeR26** [comment]
But we have still something async in the constructor, that is the root cause of PR 1245 creation.

**Comment by tabaktoni** [comment]
The imminent Issue is the wallet popup opening without the option to stop it. 
This resolves it without breaking changes so it could be merged asap.

The more proper solution could go into the next mayor versions with breaking changes.

**Review by penovicp — APPROVED** [review]
Left two suggestions that truncate the new code. Aside from that lgtm.

**Review by PhilippeR26 — COMMENTED** [review]
I think that this solution is the best compromise, even if we don't solve the initial problem without braking changes.

For Information, OKX wallet team is reporting that they have problems with this subject.
They are waiting an new version with `connect`.
So, we should go forward on this subject.

File: src/wallet/account.ts [hunk 1]
```diff
@@ -64,17 +66,11 @@ export class WalletAccount extends Account implements AccountInterface {
...
-          silent_mode: false,
-        },
-      })
-      .then((res) => {
-        this.address = res[0].toLowerCase();
+    if (!address.length) {
+      requestAccounts(this.walletProvider).then(([accountAddress]) => {
+        this.address = accountAddress.toLowerCase();
       });
+    }
```
Comment (dhruvkelawala) [inline src/wallet/account.ts:73]: Let's add a deprecation warning here saying that connect should be use instead.

File: src/wallet/account.ts [hunk 2]
```diff
@@ -170,5 +166,15 @@ export class WalletAccount extends Account implements AccountInterface {
     return signMessage(this.walletProvider, typedData);
   }
 
+  static async connect(
+    provider: ProviderInterface,
+    walletProvider: StarknetWalletProvider,
+    cairoVersion?: CairoVersion,
+    silentMode: boolean = false
```
Comment (dhruvkelawala) [inline src/wallet/account.ts:173]: I like my implementation better. Separating "connect" and "connectSilent" is a better approach imo, then passing a boolean here

Comment (tabaktoni) [reply src/wallet/account.ts:173]: OK will add connectSilent as alias on fixed connect, I think these best compromise on this.

**Comment by dhruvkelawala** [comment]
If it's based on my PR, I would also suggest changing the base branch to my PR branch and then merge this PR as "Backwards compatibility" or something.

File: src/wallet/account.ts (hunk 1 above)
Comment (tabaktoni) [reply src/wallet/account.ts]: c83e304daa58d97b0e4abac6c06f03fb3ba1652e

File: src/wallet/account.ts (hunk 2 above)
Comment (tabaktoni) [reply src/wallet/account.ts:191]: c83e304daa58d97b0e4abac6c06f03fb3ba1652e

**Pull request merged**